ClubManagement/ClubManagement/Activities/RegisterActivity.cs

```
@@ -51,24 +51,34 @@ private void SignUp(object s, EventArgs e)
 
             new Thread(() =>
             {
-                var userEmails = usersController.Values.Select(x => x.Email).ToList();
-
-                if (userEmails.Contains(edtEmail.Text))
+                try
                 {
-                    RunOnUiThread(() =>
+                    var userEmails = usersController.Values.Select(x => x.Email).ToList();
+
+                    if (userEmails.Contains(edtEmail.Text))
                     {
-                        Toast.MakeText(this, Resources.GetString(Resource.String.exist_email), ToastLength.Short).Show();
-                        dialog.Dismiss();
+                        RunOnUiThread(() =>
```

You can throw an error here. It will be catched and show the Toast message to user on the `catched` statement